margin/margin_frontend/src/ui/core/trade/slider.tsx

```diff
@@ -7,102 +7,109 @@ interface SliderProps {
-
     const canvas = canvasRef.current;
+    if (!canvas) return;
+
     const ctx = canvas.getContext("2d");
-    if (!ctx) {
-      return;
-    }
+    if (!ctx) return;
 
     const rect = canvas.getBoundingClientRect();
-
     canvas.width = rect.width;
     canvas.height = rect.height;
 
+    const xPadding = 18;
+    const totalTicks = 90;
+    const gap = (canvas.width - 2 * xPadding) / totalTicks;
+
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
     ctx.font = "10px serif";
     ctx.fillStyle = "#556571";
-    ctx.clearRect(0, 0, canvas.width, canvas.height);
-    const x = 18;
 
-    const gap = (canvas.width - x - x) / 90;
-    for (let i = 0; i <= 90; ++i) {
-      if (i % 10 == 0) {
-        if (i + 10 == Math.floor(current / 10.0) * 10) {
+    for (let i = 0; i <= totalTicks; ++i) {
+      const x = xPadding + i * gap;
+
+      if (i % 10 === 0) {
+        if ((i + 10) === Math.floor(current / 10) * 10) {
           ctx.fillStyle = "#fff";
         }
-        ctx.fillText(String(i / 10 + 1), x + i * gap - 2, 22);
+        ctx.fillText(String(i / 10 + 1), x - 2, 22);
         ctx.fillStyle = "#556571";
-        ctx.fillRect(x + i * gap, 0, 2, 10);
+        ctx.fillRect(x, 0, 2, 10);
       } else {
-        ctx.fillRect(x + i * gap, 0, 2, 4);
+        ctx.fillRect(x, 0, 2, 4);
       }
     }
 
+    // Draw the current value marker
+    const currentIndex = current - 10;
+    const markerX = xPadding + currentIndex * gap;
+
     ctx.beginPath();
-    ctx.arc(current * gap - 28, 3, 3, 0, 2 * Math.PI);
+    ctx.arc(markerX, 3, 3, 0, 2 * Math.PI);
     ctx.fillStyle = "#00D1FF";
     ctx.fill();
   }
 
   useEffect(() => {
-    rangeRef.current!.value = "10";
-    sliderChanged(10);
+    if (rangeRef.current) {
+      rangeRef.current.value = String(value);
+      sliderChanged(value);
+    }
   }, []);
 
   return (
-    <div className="w-full relative flex flex-col items-center">
-      <div
-        className={classBuilder(
-          className,
-          "w-full h-[12px] relative bg-[#01060D] rounded-full",
-        )}
-      >
-        <div
-          className={`absolute rounded-full  bg-[#00D1FF] 
-                left-[0px] right-[0px]
-                 top-0 bottom-0 inset-shadow-[#171E2852]/32 pointer-events-none z-8`}
-          style={{ right: `calc(${100 - Number(multiplayer) * 10}%  + 25px)` }}
-        ></div>
-
+      <div className="w-full relative flex flex-col items-center">
         <div
-          className="no-select absolute -top-[2px] z-10 text-[#06336E] text-[10px] font-semibold"
-          style={{ left: `calc(${_value}% - 29px - ${_value * 0.05}px)` }}
+            className={classBuilder(
+                className,
+                "w-full h-[12px] relative bg-[#01060D] rounded-full"
+            )}
         >
-          {multiplayer}
+          <div
+              className="absolute rounded-full bg-[#00D1FF] z-8"
+              style={{
+                right: `calc(${100 - value}% - 10px)`,
```

Comment:
```suggestion
                right: `calc(${100 - value}% + 5px)`
```
small adjustment to slider